cmds/unassign.go

```diff
@@ -0,0 +1,89 @@
+package cmds
+
+import (
+	"fmt"
+
+	"github.com/Betterment/testtrack-cli/fakeassignments"
+	"github.com/Betterment/testtrack-cli/schema"
+	"github.com/Betterment/testtrack-cli/validations"
+	"github.com/spf13/cobra"
+)
+
+var unassignDoc = `
+Removes split assignment from all visitors in the testtrack fake server.
+
+This command can be used to remove the assignment for a specific split
+or to remove all split assignments.
+
+Example:
+
+testtrack unassign my_fancy_experiment
+testtrack unassign --all
+`
+
+var unassignVariant string
+var unassignAll bool
+
+func init() {
+	unassignCmd.Flags().BoolVar(&noPrefix, "no-prefix", false, "Don't prefix split with app_name (supports legacy splits)")
+	unassignCmd.Flags().BoolVar(&unassignAll, "all", false, "Unassign all splits")
+	rootCmd.AddCommand(unassignCmd)
+}
+
+var unassignCmd = &cobra.Command{
+	Use:   "unassign [split_name]",
+	Short: "Remove a split assignment",
+	Long:  unassignDoc,
+	Args:  cobra.RangeArgs(0, 1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		name := ""
+		if len(args) == 1 {
+			name = args[0]
+		}
+		return unassign(name, unassignAll)
+	},
```

Comment:
i'm having an issue where `unassign` with no args is not printing out the errors. what am i doing wrong here? it seems almost like because `0` args is a valid number of args, it's ignoring the `error` that must be getting returned  from `runUnassign(name)`. or there's something obvious that i'm not seeing.

Reply:
separately, i guess i should add a validation that the name arg and all flag are mutually exclusive.

Reply:
what code branch do you expect to error if you provide no args? I don't see one.

Reply:
i'd expect `	err = validations.AutoPrefixAndValidateSplit("split_name", &name, currentAppName, mergedSchema, noPrefix, false)` to error with the split name is not a valid split. no?

Reply:
so, it seems like passing `foo` into that method doesn't error when it definitely should

Reply:
```golang
func AutoPrefixAndValidateSplit(paramName string, value *string, currentAppName string, schema *serializers.Schema, noPrefix, force bool) error {
	prefix := appNamePrefix(value)
	splitAppName := prefix
	if prefix == nil && !noPrefix {
		prefixed := fmt.Sprintf("%s.%s", currentAppName, *value)
		err := SplitExistsInSchema(paramName, &prefixed, schema)
		if err == nil {
			splitAppName = &currentAppName
			*value = prefixed
		}
    // THIS IS WHERE I'M ENDING UP with err != nil
	}
	if splitAppName != nil && *splitAppName == currentAppName && !force {
		err := SplitExistsInSchema(paramName, value, schema)
		if err != nil {
			return errors.Wrap(err, "use --force to skip schema validation")
		}
	}
	return nil
}
```

Reply:
@jmileham I haven't been able to figure out the right path here. Would love some of your help

Reply:
I think there's 2 things: one, it should validate presence at the beginning. And two, that path I'm ending up down needs a resolution

Reply:
Since there aren't tests for all the various behaviors that depend on this, it's hard to feel confident that changing this function won't have unintended side effects.